### PR TITLE
Change DB Manager Clap usage to derive

### PR DESCRIPTION
**Pull request by eserilev**
partially #5900

This is part of a series of PRs that aims to move the Lighthouse cli to clap derive. This PR moves the db manager crate to clap derive and introduces the ability to include clap derive subcommands within our existing clap builder flag definitions.

### CODE REVIEW

File: database_manager/src/lib.rs [hunk 1]
```diff
@@ -324,16 +109,15 @@ pub struct InspectConfig {
...
-fn parse_inspect_config(cli_args: &ArgMatches) -> Result<InspectConfig, String> {
-    let column = clap_utils::parse_required(cli_args, "column")?;
-    let target = clap_utils::parse_required(cli_args, "output")?;
-    let skip = clap_utils::parse_optional(cli_args, "skip")?;
-    let limit = clap_utils::parse_optional(cli_args, "limit")?;
-    let freezer = cli_args.get_flag("freezer");
-    let blobs_db = cli_args.get_flag("blobs-db");
+fn parse_inspect_config(inspect_config: &Inspect) -> Result<InspectConfig, String> {
+    let column: DBColumn = inspect_config.column.parse().unwrap();
+    let target: InspectTarget = inspect_config.output.parse().unwrap();
```
Comment (michaelsproul) [inline database_manager/src/lib.rs]: We should use errors here rather than unwrapping

**Comment by michaelsproul** [comment]
Noticed that e.g. `lighthouse db inspect --help` doesn't work. We might want to declare the help flag as a global flag or something?

**Comment by eserilev** [comment]
nice catch, [897bd8d](https://github.com/sigp/lighthouse/pull/5898/commits/897bd8d399fbe0449b2f80d6d66bbb46cfdef84f) defines the help flag as global now. this seems to have fixed the issue

File: database_manager/src/lib.rs (hunk 1 above)
Comment (eserilev) [reply database_manager/src/lib.rs]: removed the unwraps in [897bd8d](https://github.com/sigp/lighthouse/pull/5898/commits/897bd8d399fbe0449b2f80d6d66bbb46cfdef84f)

File: database_manager/src/cli.rs [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+pub struct Inspect {
+    #[clap(
+        long,
+        value_name = "TAG",
+        help = "3-byte column ID (see `DBColumn`)",
+        display_order = 0
+    )]
+    pub column: String,
+
+    // TODO InspectTarget::VARIANTS
```
Comment (michaelsproul) [inline database_manager/src/cli.rs]: should we solve this TODO before merging?

Comment (eserilev) [reply database_manager/src/cli.rs]: resolved in a32afe8

**Review by dapplion — APPROVED** [review]
LGTM! Looking forward to see all CLIs with derive :)

**Comment by jimmygchen** [comment]
@mergify queue

**Pull request merged**